src/task/block_on.rs

```diff
@@ -69,12 +69,11 @@ where
         let future = task_local::add_finalizer(future);
 
         let future = async move {
```

Comment:
For @stjepang: This is not an equivalent transform. Reading the surrounding code, I also don't think that matters, but can you please revalidate?

Reply:
This change looks fine by me. It doesn't really matter because `res` is `()`.